R/tbl_ae_rates_by_grade.R

```diff
@@ -0,0 +1,387 @@
+#' @param grade_groups (`list`)\cr
+#'   A list of formulas each corresponding to a grade group for which rates should be calculated. Grade groups must be
+#'   mutually exclusive, i.e. each grade cannot be assigned to more than one grade group. To specify each grade group,
+#'   create a formula with a vector of grades included on the left-hand side of the formula and the name of the grade
+#'   on the right-hand side, e.g. `c("1", "2") ~ "Grade 1-2"`.
+#' @param grades_exclude (`character`)\cr
+#'   A vector of grades to omit individual rows for when printing the table. These grades will still be used when
+#'   computing overall totals and grade group totals. For example, to avoid duplication, if a grade group is defined as
+#'   `"5" ~ "Grade 5"`, the individual rows corresponding to grade 5 can be excluded by setting `grades_exclude = "5"`.
+#'
+#' @returns a gtsummary table
+#' @name tbl_ae_rates_by_grade
+#'
+#' @examples
+#' ADSL <- cards::ADSL |> mutate(TRTA = ARM)
+#' ADAE_subset <- cards::ADAE |>
+#'   dplyr::filter(
+#'     AESOC %in% unique(cards::ADAE$AESOC)[1:5],
+#'     AETERM %in% unique(cards::ADAE$AETERM)[1:10]
+#'   )
+#'
+#' ## Add AETOXGR variable to example dataset
+#' set.seed(1)
+#' ADAE_subset <- ADAE_subset |>
+#'   dplyr::rowwise() |>
+#'   mutate(
+#'     AETOXGR = dplyr::case_when(
+#'       AESEV == "MILD" ~ sample(1:2, 1),
+#'       AESEV == "MODERATE" ~ sample(3:4, 1),
+#'       AESEV == "SEVERE" ~ 5,
+#'     ) |> factor(levels = 1:5)
+#'   ) |>
+#'   dplyr::ungroup()
+#'
+#' # Example 1 ----------------------------------
+#' grade_groups <- list(
+#'   c("1", "2") ~ "Grade 1-2",
+#'   c("3", "4") ~ "Grade 3-4",
+#'   "5" ~ "Grade 5"
+#' )
+#'
+#' tbl_ae_rates_by_grade(
+#'   ADAE_subset,
+#'   grade = AETOXGR,
+#'   ae = AEDECOD,
+#'   soc = AEBODSYS,
+#'   denominator = ADSL,
+#'   by = TRTA,
+#'   grade_groups = grade_groups,
+#'   grades_exclude = "5"
+#' )
+NULL
+
+#' @export
+#' @rdname tbl_ae_rates_by_grade
+tbl_ae_rates_by_grade <- function(data,
+                                  grade,
+                                  ae,
+                                  soc,
+                                  denominator,
+                                  by = NULL,
+                                  id = "USUBJID",
+                                  include_overall = all_of(c(soc, ae)),
+                                  statistic = everything() ~ "{n} ({p}%)",
+                                  label = list(
+                                    "MedDRA System Organ Class",
+                                    "MedDRA Preferred Term",
+                                    "Grade"
+                                  ) |> stats::setNames(c(soc, ae, grade)),
+                                  digits = NULL,
+                                  sort = "descending",
+                                  filter = NULL,
+                                  add_overall = FALSE,
```

Comment:
The `add_overall` argument has a default header of `"**All Active Treatments** \nN = {style_number(N)}"`. But this makes it sound like Placebo is excluded, but we're just adding a vanilla overall column.

Reply:
I'll update this - I'm not sure why but this is the label given in the GDSR.

Reply:
Users can now specify the label via `add_overall()`